### PR TITLE
migrate source GA connectors to per-stream states (2)

### PR DESCRIPTION
**Pull request by alafanechere**
## What
The following GA connectors were not yet migrated to per-stream states:
* source-twilio
* ~~source-snapchat-marketing~~
* source-google-sheets
* source-freshdesk

### CODE REVIEW

**Comment by alafanechere** [comment]
🐞 snapchat-marketing: expired credentials
🐞 twilio: expected records does not match

File: airbyte-integrations/connectors/source-google-sheets/Dockerfile [hunk 1]
```diff
@@ -34,5 +34,5 @@ COPY google_sheets_source ./google_sheets_source
 ENV AIRBYTE_ENTRYPOINT "python /airbyte/integration_code/main.py"
 ENTRYPOINT ["python", "/airbyte/integration_code/main.py"]
 
-LABEL io.airbyte.version=0.2.18
+LABEL io.airbyte.version=0.2.19
```
Comment (brianjlai) [inline airbyte-integrations/connectors/source-google-sheets/Dockerfile:37]: State isn't used by google-sheets, but for consistency, we might also want to update the overwritten `read()` method to the new signature:
```
    def read(
        self,
        logger: logging.Logger,
        config: Mapping[str, Any],
        catalog: ConfiguredAirbyteCatalog,
        state: Union[List[AirbyteStateMessage], MutableMapping[str, Any]] = None
    ) -> Generator[AirbyteMessage, None, None]:
```

File: airbyte-integrations/connectors/source-freshdesk/integration_tests/abnormal_state.json [hunk 2]
```diff
@@ -1,35 +1,79 @@
-{
-  "agents": {
-    "updated_at": "2121-11-01T00:00:00Z"
+[
+  {
+    "type": "STREAM",
+    "stream": {
+      "stream_state": { "updated_at": "2121-11-01T00:00:00Z" },
```
Comment (brianjlai) [inline airbyte-integrations/connectors/source-freshdesk/integration_tests/abnormal_state.json:5]: nit: adding multi level for readability
```
"stream_state": {
  "updated_at": "2121-11-01T00:00:00Z"
},
```

**Review by brianjlai — APPROVED** [review]
Upgrades look good, but I think it's work updating sheets with the new method signature for consistency, beyond that everything looks good! the twilio expected records failure is interesting, I wonder if this was already failing since per-stream should only affect incrementals not full refresh that expected records checks for.

**Comment by alafanechere** [comment]
/publish connector=connectors/source-freshdesk run-tests=false

**Pull request merged**